server/server/src/cold_chain.rs

```diff
@@ -0,0 +1,141 @@
+) -> Result<HttpResponse> {
+    let response = match do_login(user_info, service_provider, auth_data).await {
+        Ok(response) => response,
+        Err(error) => HttpResponse::InternalServerError().body(format!("{:#?}", error)),
+    };
+
+    Ok(response)
+}
+
+async fn do_login(
+    user_info: web::Json<LoginRequest>,
+    service_provider: Data<ServiceProvider>,
+    auth_data: Data<AuthData>,
+) -> Result<HttpResponse, RepositoryError> {
+    let service_context = service_provider.basic_context()?;
+    let sync_settings = service_provider
+        .settings
+        .sync_settings(&service_context)?
+        .unwrap();
+    let response = match LoginService::login(
+        &service_provider,
+        &auth_data,
+        LoginInput {
+            username: user_info.username.clone(),
+            password: user_info.password.clone(),
+            central_server_url: sync_settings.url.clone(),
+        },
+        MIN_ERR_RESPONSE_TIME_SEC,
+    )
+    .await
+    {
+        Ok(token) => {
+            match validate_access(
+                &service_provider,
+                &service_context,
+                &auth_data,
+                Some(token.token.clone()),
+            ) {
+                Ok(_) => {
+                    let cookie = Cookie::build("coldchain", token.token)
+                        .path(URL_PATH)
+                        .secure(true)
+                        .http_only(true)
+                        .finish();
+                    HttpResponse::Ok()
+                        .append_header(header::ContentType(mime::APPLICATION_JSON))
+                        .cookie(cookie)
+                        .body(r#"{ "success": true }"#)
+                }
+                Err(error) => {
+                    let formatted_error = format!("{:#?}", error);
+                    HttpResponse::Unauthorized().body(formatted_error)
+                }
+            }
+        }
+        Err(error) => {
+            let formatted_error = format!("{:#?}", error);
+            HttpResponse::Unauthorized().body(formatted_error)
+        }
+    };
+
+    Ok(response)
+}
+
+/// Validates current user is authenticated and authorized
+pub fn validate_access(
+    service_provider: &ServiceProvider,
+    service_context: &ServiceContext,
+    auth_data: &AuthData,
+    token: Option<String>,
+) -> Result<ValidatedUser, AuthError> {
+    let user_service = UserAccountService::new(&service_context.connection);
+    let validated_user = validate_auth(auth_data, &token)?;
+    let store_id = match user_service.find_user(&validated_user.user_id)? {
+        Some(user) => {
+            let store_id = match user.default_store() {
+                Some(store) => Some(store.store_row.id.clone()),
+                None => None,
+            };
+            store_id
+        }
+        None => {
+            return Err(AuthError::Denied(AuthDeniedKind::NotAuthenticated(
+                "No default store".to_string(),
```

Comment:
this is user not found error, I think the default store error should be up one and read something like "No default store that is active on current site", and the `None` return for user would be internal error

Reply:
good point - thanks, have updated